src/lib.rs

```diff
@@ -112,24 +116,37 @@ pub(crate) fn calc_nd1nd2(inputs: &Inputs) -> Result<(f32, f32), String> {
 /// Calculates the n probability density function (PDF) for the given input.
 /// # Returns
 /// f32 of the value of the n probability density function.
-pub(crate) fn calc_npdf(x: f32) -> f32 {
-    let d: f32 = (x - N_MEAN) / N_STD_DEV;
-    (-HALF * d * d).exp() / (SQRT_2PI * N_STD_DEV)
+pub(crate) fn calc_npdf<T>(x: T) -> T
+where
+    T: Float + FromPrimitive,
+{
+    let n_mean = T::from(N_MEAN).unwrap();
+    let n_std_dev = T::from(N_STD_DEV).unwrap();
+    let sqrt_2pi = T::from(SQRT_2PI).unwrap();
+    let half = T::from(HALF).unwrap();
+    let d: T = (x - n_mean) / n_std_dev;
+    (-half * d * d).exp() / (sqrt_2pi * n_std_dev)
 }
 
 /// # Returns
 /// f32 of the derivative of the nd1.
-pub fn calc_nprimed1(inputs: &Inputs) -> Result<f32, String> {
+pub fn calc_nprimed1<T>(inputs: &Inputs<T>) -> Result<T, String>
+where
+    T: Float + FromPrimitive,
+{
     let (d1, _) = calc_d1d2(inputs)?;
 
     // Get the standard n probability density function value of d1
-    let nprimed1 = calc_npdf(d1);
+    let nprimed1 = calc_npdf(From::from(d1));
```

Comment:
i think 
```suggestion
    let nprimed1 = calc_npdf(d1);
```
it is enough.